src/main/java/com/example/demo/services/LeaseManagementDropBoxImpl.java

```diff
@@ -38,22 +43,26 @@ public class LeaseManagementDropBoxImpl implements LeaseManagement {
 
     @Autowired
     public LeaseManagementDropBoxImpl(UserRepository userRepository, LeaseRepository leaseRepository, ApartmentRepository apartmentRepository, TenantRepository tenantRepository) {
-        ApiClient apiClient = com.dropbox.sign.Configuration.getDefaultApiClient().setApiKey(dropBoxSignToken);
-        this.signatureRequestApi = new SignatureRequestApi(apiClient);
         this.userRespository = userRepository;
         this.leaseRepository = leaseRepository;
         this.apartmentRepository = apartmentRepository;
         this.tenantRepository = tenantRepository;
     }
 
+    @PostConstruct
+    private void initApiClient() {
+        ApiClient apiClient = com.dropbox.sign.Configuration.getDefaultApiClient().setApiKey(dropBoxSignToken);
+        this.signatureRequestApi = new SignatureRequestApi(apiClient);
+    }
+
 
     @Transactional(rollbackFor = Exception.class)
     public SignatureRequestGetResponse createLeaseSignatureRequest(LeaseSignRequestDTO leaseSignRequestDTO) throws ApiException {
```

Comment:
It is a bit hard to differentiate what happens here. I see that you use a lot of mock data . Is it possible to use the data in the database and remove all the mocking of the data from the function? 

And for my understandig, is this the first time a user is going to sign a lease contract? Or is it an extention of the contract? WIth other words, are the users theoretically alreading in the system or not?

Reply:
if I am correct here, there are happining two things. Getting all the data to send a signature request and sending the request. Besides that, there is the collection of data for saving the lease into the database and saving it. Maybe you could seperate the two in different functions for clarity?